src/pages/components/WebView.qml

```diff
@@ -278,35 +276,12 @@ WebContainer {
                 }
 
                 switch (message) {
-                case "chrome:linkadded": {
-                    var parsedFavicon = false
-                    var acceptedTouchIcon = (iconType === "apple-touch-icon" || iconType === "apple-touch-icon-precomposed")
-                    var acceptableTouchIcon = (data.rel === "apple-touch-icon" || data.rel === "apple-touch-icon-precomposed")
-                    if (data.href && (data.rel === "icon" || acceptableTouchIcon)) {
-                        var sizes = []
-                        if (data.sizes) {
-                            var digits = data.sizes.split("x")
-                            var size = digits && digits.length > 0 && digits[0]
-                            if (size) {
-                                sizes.push(size)
-                            }
-                        }
-
-                        for (var i in sizes) {
-                            var faviconSize = parseInt(sizes[i])
-                            // Accept largest icon but one that is still smaller than Theme.itemSizeExtraLarge.
-                            if (faviconSize && faviconSize > iconSize && faviconSize <= Theme.itemSizeExtraLarge * 2) {
-                                iconSize = faviconSize
-                                parsedFavicon = true
-                            }
-                        }
-                    }
+                case "Link:SetIcon": {
+                  if (acceptedTouchIcon)
+                        return
 
-                    // Always pick at least one touch icon, parsed favicons are always based on size (good enough quality)
-                    if (!acceptedTouchIcon && acceptableTouchIcon || parsedFavicon) {
-                        favicon = data.href
-                        iconType = iconSize >= Theme.iconSizeMedium ? data.rel : ""
-                    }
+                    acceptedTouchIcon = data.isRichIcon
```

Comment:
Is the idea for using the `acceptedTouchIcon` flag that it forces use of the rich icon in preference to all others?

Reply:
Yes but there's necessarily no rich icon.